include/hyprutils/animation/AnimationManager.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "./BezierCurve.hpp"
+#include "./AnimatedVariable.hpp"
+#include "../math/Vector2D.hpp"
+#include "../memory/SharedPtr.hpp"
+
+#include <unordered_map>
+#include <vector>
+
+using namespace Hyprutils::Memory;
+using namespace Hyprutils::Math;
+
+namespace Hyprutils {
+    namespace Animation {
+
+        /* A class for managing bezier curves and variables that are being animated. */
+        class CAnimationManager {
+          public:
+            CAnimationManager();
+
+            void                                                          tickDone();
+            bool                                                          shouldTickForNext();
+
+            virtual void                                                  scheduleTick() = 0;
+            virtual void                                                  onTicked()     = 0;
+
+            void                                                          addBezierWithName(std::string, const Vector2D&, const Vector2D&);
+            void                                                          removeAllBeziers();
+
+            bool                                                          bezierExists(const std::string&);
+            CSharedPointer<CBezierCurve>                                  getBezier(const std::string&);
+
+            std::unordered_map<std::string, CSharedPointer<CBezierCurve>> getAllBeziers();
+
+            std::vector<CBaseAnimatedVariable*>                           m_vActiveAnimatedVariables;
```

Comment:
same 'ere, can we make this memory safe?

Reply:
Since we declare animated variables as members, we would need smth like a separate variable container class, so that the AnimationManager does not store a reference to the member, but to a SP created by the member.
Or we actually store `SP<CAnimatedVariable<float>>` where they are used.
With both options operator= has to be ditched i feel like and the code where the vars are used needs to be updated.

Reply:
`SP<CAnimVariable<type>>` was what I was thinking bout.